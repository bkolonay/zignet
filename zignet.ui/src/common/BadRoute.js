import React, { Component } from 'react';

class BadRoute extends Component {

  render() {
    return (
      <p>Route does not exist</p>
    );
  }
}

export default BadRoute;
