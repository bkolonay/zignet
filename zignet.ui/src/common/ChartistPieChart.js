import React, { Component } from 'react';
import Chartist from 'chartist';
import './chartist.css';

class ChartistPieChart extends Component {

  componentDidMount() {
  	this.chartistChart = Chartist.Pie(
  		this.chartContainer,
  		this.props.chartData,
  		{ 
  			donut: true,
			donutWidth: 40
		}
	)
  }

  render() {
    return (
      <div 
        className="ct-chart" 
        id={"chart" + this.props.chartId}
        ref={(div) => this.chartContainer = div}
      />
    );
  }
}

export default ChartistPieChart;
