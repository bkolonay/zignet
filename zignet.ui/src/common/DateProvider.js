import moment from 'moment';

function now() {
	return moment();
}

export { now }