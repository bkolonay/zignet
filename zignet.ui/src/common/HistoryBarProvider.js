import moment from 'moment';

class HistoryBarProvider {

  /* 
    helped with percentage calculations:
     - https://www.calculatorsoup.com/calculators/math/percentage.php
     - https://www.timeanddate.com/date/timeduration.html
  */
  
	getErrorDivAttributes(historyBarWidth, now, testFailureDuration) {
		let oneDayAgo = moment(now.format()).subtract(24, 'hours');
    let testFailureStartTime = moment.utc(testFailureDuration.FailureStart).local();
    let minutesBetweenOneDayAgoAndTestFailureStartTime = moment.duration(testFailureStartTime.diff(oneDayAgo)).asMinutes();

    var failureDivStart = 0;
    if (minutesBetweenOneDayAgoAndTestFailureStartTime > 0) {
      let passingPercentOfLastDay = (minutesBetweenOneDayAgoAndTestFailureStartTime / 1440) * 100;
      failureDivStart = Math.floor((passingPercentOfLastDay / 100) * historyBarWidth);
    }
    else
      testFailureStartTime = oneDayAgo;

    var failureDivWidth = 0;
    if (testFailureDuration.FailureEnd) {
      let testFailureEndTime = moment.utc(testFailureDuration.FailureEnd).local();
      let failureDurationInMinutes = moment.duration(testFailureEndTime.diff(testFailureStartTime)).asMinutes();
      let failingPercentOfLastDay = (failureDurationInMinutes / 1440) * 100;
      failureDivWidth = Math.floor((failingPercentOfLastDay / 100) * historyBarWidth);
      if (failureDivWidth < 1)
      	failureDivWidth = 1;
    }
    else 
      failureDivWidth = historyBarWidth - failureDivStart;

    return {left: failureDivStart, width: failureDivWidth};
	}

  getDivTitle(testFailureDuration) {
    let testFailureStartTime = moment.utc(testFailureDuration.FailureStart).local();
    if (testFailureDuration.FailureEnd)
    {
      let testFailureEndTime = moment.utc(testFailureDuration.FailureEnd).local();
      let timeFromStartToEnd = testFailureStartTime.to(testFailureEndTime, true);
      return timeFromStartToEnd + ' from ' + testFailureStartTime.format('l h:mma') + ' - ' + testFailureEndTime.format('l h:mma');
    }
    else
    {
      let timeFromStartToEnd = testFailureStartTime.toNow(true);
      return timeFromStartToEnd + ' from ' + testFailureStartTime.format('l h:mma') + ' - now';
    }      
  }
}

export default HistoryBarProvider;