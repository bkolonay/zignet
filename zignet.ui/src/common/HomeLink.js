import React, { Component } from 'react';
import { getUrlParams } from '../common/routing/SearchFilter.js'

class HomeLink extends Component {

  render() {
    return (
      <a href={"/?" + getUrlParams(this.props.filter)}>ZigNet</a>
    );
  }
}

export default HomeLink;
