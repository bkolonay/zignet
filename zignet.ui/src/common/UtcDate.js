import moment from 'moment';

class UtcDate {
  constructor(date) {
    this.moment = moment.utc(date);
  }

  toLocal() {
    this.moment = this.moment.local();
    return this;
  }

  getDateAndTimeString() {
    return this.moment.format('L LTS');
  }

  getTimeFromNow() {
    return this.moment.fromNow(true);
  }

  getTimeFromNowWithSuffix() {
    return this.moment.fromNow();
  }  

  getTimeFrom(date) {
    const localMoment = moment.utc(date);
    return this.moment.from(localMoment, true);
  }  
}

export default UtcDate;
