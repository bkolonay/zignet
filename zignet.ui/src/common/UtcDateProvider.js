import moment from 'moment';

function getTimeFromNowWithSuffix(utcDateTime) {
	let localMoment = moment.utc(utcDateTime);
	return localMoment.fromNow();
}

export { getTimeFromNowWithSuffix }