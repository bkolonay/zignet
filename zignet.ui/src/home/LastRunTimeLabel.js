import React, { Component } from 'react';
import { getTimeFromNowWithSuffix } from '../common/UtcDateProvider'

class LastRunTimeLabel extends Component {

  render() {

    let label = null;
    if (this.props.grouped)
    	label = <p/>;
    else if (this.props.suiteEndTime)
    	label = <p className="text-center text-muted"><small>{getTimeFromNowWithSuffix(this.props.suiteEndTime)}</small></p>;
    else
    	label = <p className="text-center text-warning"><small>running...</small></p>;

    return (
      label
    );
  }
}

export default LastRunTimeLabel;
