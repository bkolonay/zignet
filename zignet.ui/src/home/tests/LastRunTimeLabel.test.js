import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import LastRunTimeLabel from '../LastRunTimeLabel';
import { getTimeFromNowWithSuffix } from '../../common/UtcDateProvider';

jest.mock('../../common/UtcDateProvider');

it('renders when not grouped', () => {
  const component = <LastRunTimeLabel grouped={false} suiteEndTime={'2018-05-24T01:00:00'} />;
  ReactDOM.render(component, document.createElement('div'));
});

it('renders when grouped', () => {
  const component = <LastRunTimeLabel grouped={true} suiteEndTime={'2018-05-24T01:00:00'} />;
  ReactDOM.render(component, document.createElement('div'));
});

it('snapshot when not grouped and has end time', () => {
  getTimeFromNowWithSuffix.mockReturnValue('3 days ago');
  const component = <LastRunTimeLabel grouped={false} suiteEndTime={'2018-05-24T01:00:00'} />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

it('snapshot when grouped', () => {
  const component = <LastRunTimeLabel grouped={true} suiteEndTime={'2018-05-24T01:00:00'} />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

it('snapshot when grouped and no suite end time', () => {
  const component = <LastRunTimeLabel grouped={true} suiteEndTime={undefined} />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

it('snapshot when not grouped and no suite end time', () => {
  const component = <LastRunTimeLabel grouped={false} suiteEndTime={undefined} />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});