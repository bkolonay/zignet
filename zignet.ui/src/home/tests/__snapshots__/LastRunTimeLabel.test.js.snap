// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot when grouped 1`] = `<p />`;

exports[`snapshot when grouped and no suite end time 1`] = `<p />`;

exports[`snapshot when not grouped and has end time 1`] = `
<p
  className="text-center text-muted"
>
  <small>
    3 days ago
  </small>
</p>
`;

exports[`snapshot when not grouped and no suite end time 1`] = `
<p
  className="text-center text-warning"
>
  <small>
    running...
  </small>
</p>
`;
